README.md

```diff
@@ -172,112 +163,144 @@ specify the class of the instance we want to access.
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.context.annotation.AnnotationConfigApplicationContext;
 
-public class Main {
-    public static void main(String[] args) {
-        // create context
-        AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext(MyConfig.class);
-        // access bean
-        Dog luna = context.getBean(Dog.class);
-        luna.setName("Luna");
-        System.out.println(luna);
-    }
+@SpringBootApplication
+public class DemoApplication {
+   public static void main(String[] args) {
+      SpringApplication.run(DemoApplication.class, args);
+
+      // create Spring context instance
+      AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext(MyConfig.class);
+
+      // access the bean
+      Dog luna = context.getBean(Dog.class);
+      luna.setName("Luna");
+      System.out.println(luna);
+   }
 }
 ```
 
 Running the `main` method will print out the `Dog` instance in the Spring
 context. The stereotype annotations are primarily used over the `@Bean`
 annotations since it is more simple.
 
-As mentioned earlier, the `org.springframework.stereotype` package has other
+So far, we have been using the `@Service` and `@Controller` annotations, which
+are part of the `org.springframework.stereotype` package, and are applied to
+classes. Both of these annotations serve as a specialization of `@Component`,
+allowing for implementation classes to be autodetected through classpath scanning.
+This is how we were able to add them as beans to the application context in the
+previous lessons!
+
+But wait... where did we scan for the components and where was the configuration
+class?
+
+## The `@SpringBootApplication` annotation
+
+We've seen how the Spring Initializr creates a default application class with the
+`@SpringBootApplication` annotation. As we mentioned before, this is the main
+entry point to our Spring Boot API. But there's more to this annotation! The
+`@SpringBootApplication` is a single annotation combining three popular
 annotations:
 
-- `@Service`: applied to classes used for business logic.
-- `@Controller`: applied to classes used in a web environment.
-- `@Repository`: applied to classes used for interacting with external data
-  storage.
+- `@EnableAutoConfiguration`: This annotation will enable Spring Boot's
+  auto-configuration mechanism. To read more about this annotation, please see
+  its documentation here: [Auto-configuration doc](https://docs.spring.io/spring-boot/docs/2.0.x/reference/html/using-boot-auto-configuration.html).
+- `@ComponentScan`: As we saw above, this annotation will scan the package where
+  the application is located for component classes (marked with `@Component`).
+- `@Configuration`: As we saw above, this annotation allows us to register beans
+  in the application context.
+
```

Comment:
A common mistake during the BlackRock program was students putting their controllers and services in packages that weren't scanned automatically.
So if the students had their main class in com.example.demo, they may have tried to put their controllers in com.example.controllers rather than com.example.demo.controllers.

Would it make sense to include a warning about that mistake somewhere in this lesson?

Reply:
Let me know if you think my explanation is a little better about the base package and the `@ComponentScan`